samcli/hook_packages/terraform/copy_terraform_built_artifacts.py

```diff
@@ -276,6 +286,31 @@ def create_backend_override():
         cli_exit()
 
 
+def validate_environment_variables():
```

Comment:
What is the scenario where these environment variables aren't set when checking at the hook level but are when running this script?

Reply:
Normally, this shouldn't happen when running `sam build` since the customer shouldn't have to manually run `sam build -t <aws sam iacs metadata folder>/template.json`, but the customer is still able to run this manually. So having it in two locations solves:
1. `sam build --hook-name` 
    - hook check covers this
3. `sam build --hook-name`, then for some reason `sam build -t <iacs metadata template>`
   - hook check covers first build run
   - script covers the second run